### PR TITLE
Accept new / unknown API fields gracefully

### LINKED ISSUE #130: Definitions should gracefully accept unknown params
From discussions in Montreal, it seems that new fields in the Juju schema definitions are expected to be handled gracefully even without a bump to the facade version.  That is, facade versions will only be bumped if a field is removed or changed.

### PR DESCRIPTION
**Pull request by johnsca**
From discussions in Montreal, it seems that new fields in the Juju schema definitions are expected to be handled gracefully even without a bump to the facade version. That is, facade versions will only be bumped if a field is removed or changed.

Fixes #130
Also fixes conjure-up/conjure-up#903

### CODE REVIEW

File: docs/contributing/index.rst [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+Once you have a new schema file, you can update the Python code
+using the `client` make target:
+
+.. code:: bash
+
+  make client
+
+You should expect to see updates to the `juju/client/_definitions.py` file,
+as well as one or more of the `juju/client/_clientX.py` files, dependning on
```
Comment (tvansteenburgh) [inline docs/contributing/index.rst]: depending

File: docs/contributing/index.rst [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+which facades were touched.
+
+
+Overrides
+---------
+
+It should be quite rare, but occasionally the generated Python code is does
```
Comment (tvansteenburgh) [inline docs/contributing/index.rst]: s/is//

File: docs/contributing/index.rst [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+not capture all of the logic needed properly parse the output from the API
```
Comment (tvansteenburgh) [inline docs/contributing/index.rst]: needed to properly

File: docs/index.rst [hunk 4]
```diff
@@ -16,6 +16,7 @@ Table of Contents
 
    narrative/index
    API Docs <api/modules>
+   schema/index
```
Comment (tvansteenburgh) [inline docs/index.rst]: Should this be contributing/index?

File: juju/client/overrides.py [hunk 5]
```diff
@@ -96,5 +101,94 @@ class AllWatcherFacade(Type):
...
+                raise TypeError('Unable to parse Binary version string: '
+                                '{}'.format(data))
+        d = {}
+        for k, v in (data or {}).items():
+            d[cls._toPy.get(k, k)] = v
+
+        return cls(**d)
+
+    def serialize(self):
+        return "%s-%s-%s" % (self.number.serialize(), self.series, self.arch)
```
Comment (tvansteenburgh) [inline juju/client/overrides.py]: Number and Binary should at least have a happy path test case for parse/serialize.

File: docs/contributing/index.rst (hunk 3 above)
Comment (battlemidget) [inline docs/contributing/index.rst]: i wasn't going to nitpick but also `Python code is does` should be fixed

**Comment by battlemidget** [comment]
I just tested a new built snap pulling in this branch for conjure-up and deployments succeed again.

**Comment by johnsca** [comment]
For reference, the tests were ported from https://github.com/juju/version/blob/master/version_test.go

**Pull request merged**